cmd/kubelet/app/server.go

```diff
@@ -186,6 +186,120 @@ func Run(s *options.KubeletServer, kubeDeps *kubelet.Dependencies) error {
 	return nil
 }
 
+// setKubeDepsClients initializes clients if not standalone mode and any of the clients are not provided
+func setKubeDepsClients(s *options.KubeletServer, kubeDeps *kubelet.Dependencies, nodeName types.NodeName) error {
+	clientConfig, err := CreateAPIServerClientConfig(s)
+	if err == nil {
```

Comment:
You can check `err != nil` first, and avoid nesting the main logic.

Reply:
I have put the check out of the new function, but in the case of s.KubeConfig.Provided(), it is just a warning, and it will not go into setKubeDepsClients.